SygicUIKit/Classes/Views/ProgressView/SYUICircleGradientProgressView.swift

```diff
@@ -0,0 +1,175 @@
+            dashedBorder.strokeEnd = progressOffset/2.0 + roundProgress*(1-progressOffset)
+        }
+    }
+    
+    /// Offsett sets size of the circle segment that is not drawn [0...1].
+    /// 0 ... whole circle is drawn, 0.5 ... only upper half of circle is drawn
+    public var progressOffset: CGFloat = 0 {
+        didSet {
+            updateProgressOffset()
+        }
+    }
+    
+    /// Gradient layer colors
+    public var gradientColors: [CGColor] = [
+        UIColor.green.cgColor,
+        UIColor.green.cgColor,
+        UIColor.green.cgColor,
+        UIColor.orange.cgColor,
+        UIColor.red.cgColor
+        ] {
+        didSet {
+            gradient.colors = gradientColors
+        }
+    }
+    
+    public let dashedBackground = CAShapeLayer()
+    public let dashedBorder = CAShapeLayer()
+    public let gradient = CAGradientLayer()
+    
+    // MARK: - Private properties
+    
+    private var dashPattern: [NSNumber] {
+        return [dashLength, dashSpace]
+    }
+    
+    // MARK: - Public methods
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+    
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+        updateLayers()
+    }
+    
+    // MARK: - Private methods
+    
+    private func setup() {
+        backgroundColor = .clear
+        setupDashShapeLayer(dashedBackground)
+        setupDashShapeLayer(dashedBorder)
+        updateProgressOffset()
+        setupGradientLayer()
+        gradient.mask = dashedBorder
+        layer.addSublayer(dashedBackground)
+        layer.addSublayer(gradient)
+    }
+    
+    private func setupDashShapeLayer(_ shapeLayer: CAShapeLayer) {
+        shapeLayer.lineDashPattern = dashPattern
+        shapeLayer.lineWidth = dashWidth
+        shapeLayer.lineCap = dashCap
+        shapeLayer.fillColor = nil
+        shapeLayer.strokeColor = UIColor.black.cgColor
+        
+        let rotatateTr = CATransform3DRotate(CATransform3DIdentity, .pi/2.0, 0, 0, 1)
+        shapeLayer.transform = rotatateTr
+    }
+    
+    private func setupGradientLayer() {
+        if #available(iOS 12.0, *) {
+            gradient.type = .conic
+        }
+        gradient.colors = gradientColors
+        gradient.startPoint = CGPoint(x: 0.5, y: 0.5)
+        gradient.endPoint = CGPoint(x: 0.5, y: 1)
+    }
+    
+    private func updateDashShapeLayer(_ shapeLayer: CAShapeLayer, frame: CGRect, path: UIBezierPath) {
+        shapeLayer.frame = frame
+        shapeLayer.path = path.cgPath
+    }
+    
+    private func updateProgressOffset() {
+        let offset = progressOffset/2.0
+        dashedBackground.strokeStart = offset
+        dashedBorder.strokeStart = offset
+        dashedBackground.strokeEnd = 1-offset
+        dashedBorder.strokeEnd = 1-offset
+        let p = progress
+        progress = p
```

Comment:
Dalo by sa tu pouzit bud `progress = { progress }()` alebo idealne v tom didSet zavolat nejaku private funkciu a tu ju volat tiez? Akoze moze to byt v podstate aj takto ale je to take trochu nie moc clean.